### ISSUE TITLE
Remove date.js functions (and possibly the file itself)

### ISSUE THREAD

**Issue by wkurniawan07** [TEAMMATES/teammates#4767]
In #4688 the function `isDate` in `date.js` is removed because it clashed with the one defined in `common.js`. It prompted me to question the very necessity of the entire `date.js` file.
I tried removing `date.js` from existence (even those pages that "import" it) and guess what? Everything works without any problem. There can only be one conclusion: `date.js` has been unused since time immemorial.

After #4688 is merged (very soon), the unused functions can be removed. In fact, the functions that are still used can be moved to `common.js` and then `date.js` can be removed altogether.

**Comment by wkurniawan07** [comment]
@jkt001 assigning to you since you told me you want to do it :p

**Comment by jkt001** [comment]
The only (currently known) functions needed from <code>date.js</code> are:
1. <code>formatDate</code>
2. <code>convertToClientTimeZone</code>
3. <code>convertToUtc</code> (for testing purposes)
4. <code>getTimeZoneStringFromDate</code>

The original one from <code>date.js</code> that is used is only <code>formatDate</code> while the others are newly-written in #4688. 

One option is to remove <code>date.js</code> and move the said methods to <code>common.js</code>.

**Comment by wkurniawan07** [comment]
Since we're keeping #4688 on hold instead of this, you can proceed if you wish.
